src/main.rs

```diff
@@ -1,3 +1,77 @@
-fn main() {
-    println!("Hello, world!");
+use game_of_life::model;
+use macroquad::{miniquad::window::screen_size, prelude::*};
+
+const GRID_WIDTH: usize = 50;
+const GRID_HEIGHT: usize = 50;
+
+fn screen_to_grid(screen_pos: (f32, f32)) -> (usize, usize) {
+    let grid_screen_ratio = (
+        GRID_WIDTH as f32 / screen_width(),
+        GRID_HEIGHT as f32 / screen_height(),
+    );
+
+    let x = (grid_screen_ratio.0 * screen_pos.0).floor();
+    let y = (grid_screen_ratio.1 * screen_pos.1).floor();
+
+    (x as usize, y as usize) // Will always be greater than or equal to 0
+}
+
+fn edit_mode(grid: &mut model::Grid) {
+    if !is_mouse_button_pressed(MouseButton::Left) {
+        return;
+    }
+
+    let (x, y) = screen_to_grid(mouse_position());
+    grid.toggle(x, y);
+}
+
+// Updates the state of the current grid given a the previous one
+fn update(prev: &model::Grid, grid: &mut model::Grid) {
+    for (x, y) in prev.coordinates() {
+        let new_state = model::rules::compute_new_state(prev, x, y);
+        grid.update(x, y, new_state);
+    }
+}
+
+fn draw_ui(is_editing: &bool) {
```

Comment:
```suggestion
fn draw_ui(is_editing: bool) {
```

Is not necessary to pass this boolean as a reference

Reply:
It is updated.